### PR TITLE
Enable spline methods on RegularGridInterpolator

### PR DESCRIPTION
**Pull request by andfoy**
See https://github.com/cupy/cupy/issues/7186

### CODE REVIEW

File: cupyx/scipy/interpolate/_rgi.py [hunk 1]
```diff
@@ -495,7 +583,8 @@ def interpn(points, values, xi, method="linear", bounds_error=True,
                                           resampling)
     """
     # sanity check 'method' kwarg
-    if method not in ["linear", "nearest"]:
+    if method not in ["linear", "nearest", "slinear", "cubic",
+                      "quintic", "pchip"]:
         raise ValueError(
             "interpn only understands the methods 'linear' and 'nearest'. "
             "You provided {method}.")
```
Comment (takagi) [inline cupyx/scipy/interpolate/_rgi.py:590]: Would you also update this message?

**Review by takagi — APPROVED** [review]
LGTM except for a minor comment!

**Review by takagi — DISMISSED** [review]

**Comment by takagi** [comment]
Jenkins test is failing on ROCm environment. Let me see it.

**Comment by takagi** [comment]
As we saw in another PR, ROCm does not have `csrlsvqr` yet so some tests are failing there. Would you skip the following tests on ROCm?

```
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_complex[slinear]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_complex[cubic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_complex[quintic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_complex[pchip]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_descending_points[cubic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_descending_points[quintic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_fill_value[cubic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_fill_value[quintic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_nonscalar_values[cubic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_nonscalar_values[quintic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_nonscalar_values_2[False-cubic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_nonscalar_values_2[False-quintic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_nonscalar_values_2[True-cubic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestRegularGridInterpolator::test_nonscalar_values_2[True-quintic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestInterpN::test_xi_broadcast[cubic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestInterpN::test_xi_broadcast[quintic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestInterpN::test_nonscalar_values[cubic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestInterpN::test_nonscalar_values[quintic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestInterpN::test_nonscalar_values_2[cubic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestInterpN::test_nonscalar_values_2[quintic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestInterpN::test_complex[slinear]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestInterpN::test_complex[cubic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestInterpN::test_complex[quintic]
FAILED cupyx_tests/scipy_tests/interpolate_tests/test_rgi.py::TestInterpN::test_complex[pchip]
```

**Comment by takagi** [comment]
Record: As for performance comparison, this PR adds `slinear`, `cubic`, `quintic`, and `pchip` methods to RegularGridInterpolator. `slinear`, `cubic`, and `quintic` methods create Bspline, and `pchip` method creates PchipInterpolator. Refer to them for their performance.

**Review by takagi — APPROVED** [review]

**Comment by ev-br** [comment]
Performance-wise, I wouldn't expect great performance, either in CuPy or in SciPy really. This class creates a tensor product spline, and the current implementation iterates over the dimensions: in D dimensions it constructs D one-dimensional spline interpolators at each call. A better implementation would require a tensor product spline object. This is on the SciPy TODO list. Once it's done in SciPy, we'll port it to CuPy, too.

**Pull request merged**